frontend/server/utils.ts

```diff
@@ -65,6 +66,114 @@ export function loadJSON<T>(filepath?: string, defaultValue?: T): T | undefined
   }
 }
 
+/**
+ * find final file path in pod:
+ * 1. check volume and volume mount exist in pod
+ * 2. if volume mount configured with subPath, check filePathInVolume startsWith subPath and prune filePathInVolume
+ * 3. concat volume mount path with pruned filePathInVolume as final path or error message if check failed
+ * @param pod contains volumes and volume mounts info
+ * @param options
+ *  - containerNames optional, will match to find container or container[0] in pod will be used
+ *  - volumeMountName container volume mount name
+ *  - filePathInVolume file path in volume
+ * @return [final file path, error message if check failed]
+ */
+export function findFileOnPodVolume(
+  pod: any,
+  options: {
+    containerNames: string[] | undefined;
+    volumeMountName: string;
+    filePathInVolume: string;
+  },
+): [string, string | undefined] {
+  const { containerNames, volumeMountName, filePathInVolume } = options;
+
+  const volumes = pod?.spec?.volumes;
+  const prefixErrorMessage = `Cannot find file "volume://${volumeMountName}/${filePathInVolume}" in pod ${pod
```

Comment:
nit: adding quotes around dynamic value makes error message easier to understand
```
  const prefixErrorMessage = `Cannot find file "volume://${volumeMountName}/${filePathInVolume}" in pod "${pod?.metadata?.name || 'unknown'}":`
```